hicommonwealth/commonwealth
# Client-side Cosmos Proposals-by-type caching with React Query

## Link to Issue
Closes: #4561 

## Description of Changes
- uses React Query to cache activeProposals and completedProposals client-side
- lazy load them in /proposals and Proposal Link Modal (from a thread)
- removes initProposals from governance init.

## Test Plan
- Ran proposals.spec E2E tests locally
- CA (click around) tested on local:
  - http://localhost:8080/osmosis/proposals
      - CA to individual proposals and back
      - Create new thread and link proposals
      - CA to other communities (kyve) and back
      - Generally expect cached results of previously visited proposals fetches (with background refreshing)
  - http://localhost:8080/csdk/proposals
      - create proposal, wait 10 minutes, click away from and back to proposals, expect inactive proposals to show new one (this requires redis caching to be turned off, otherwise it will take an extra 5 minutes).
      - demonstrates cache background refreshing

## Deployment Plan

1. 

## Other Considerations

- The initial load of /proposals page can still show "Starting Chain" for a bit because it waits for the API to initialize. Hoping to improve this in #4527
